python/BioSimSpace/Process/_gromacs.py

```diff
@@ -2255,7 +2262,7 @@ def _add_position_restraints(self, config):
 
                 # Write the updated topology to file.
                 with open(self._top_file, "w") as file:
-                    for line in top_lines:
+                    for line in new_top_lines:
                         file.write("%s\n" % line)
 
             # A user-defined list of atoms indices.
```

Comment:
Is this issue also not present in this branch of the conditional?

Reply:
Ah yes, good spot. Will update this now.

Reply:
This should now be resolved. When I get a chance I might try to consolidate the two branches with a function call. The logic used to be quite a bit more different, but I managed to adjust things a while ago so the loops are effectively the same, although use different data structures.